### ISSUE TITLE
Doesn't work if the target branch name contains a dot (.) character

### ISSUE THREAD

**Issue by ITMATIC-HUSO** [Jezusfan/MultiMerge#2]
First of all, thanks for the amazing extension!

Our branches are named as follows: BranchName.Net

If we try to unshelve to another branch, the drop down shows the following options:
$BranchName.
$OtherBranchName.

Should be:
$/BranchName.Net
$/OtherBranchName.Net

If we manually append ".Net" into the input box, the shelvesets ends up at the folder $/BranchName.Net/Net which creates a new folder called "Net" inside the target branch and results in all files in the shelveset being marked as "Add".

Any workarounds available?

**Comment by Jezusfan** [comment]
Hi, I've tried your scenario. I have created a Team Project, called 'Wigman'. In that project I have created a branch called BranchName.Net. But the path it creates by default is $/Wigman/BranchName.Net. For some reason, you seem to be missing the Team Project-name? When I create a branch off my first branch, called OtherBranchName.Net, I have no trouble merging between branches (and it auto-populates the branch-names correctly in the dropdown). Perhaps I am misunderstanding your scenerio?


![image](https://user-images.githubusercontent.com/7585731/212023370-519accf9-7cf9-4dff-90db-718f1f9403d9.png)

**Comment by ITMATIC-HUSO** [comment]
Thanks for the quick reply. This is what I get:

![image](https://user-images.githubusercontent.com/60346957/212026477-83a39942-b444-4e00-97cc-cfa3bbf79034.png)

(I'm at the "Find Shelvesets" screen)

**Comment by ITMATIC-HUSO** [comment]
FYI: After seeing your screenshot, I tried the "Marge ChangeSet by Comment", which indeed seems to work. So issue is only present if trying to unshelve shelvesets to another branch.

**Comment by Jezusfan** [comment]
Hmm, I get a 'could not determine workspace'-exception when trying to unshelve to another branch. I'll try to dig into what's causing that, and see if I can replay your scenario afterwards. Perhaps the dot (.) in your projectname causes an issue. Thanks for the details so far, I'll check it out and let you know!

**Comment by ITMATIC-HUSO** [comment]
I finally have the project up for debugging and found out that the issue is in the `MergeWorker.GetWorkSpace` function. Both the target versions as well as the source version returns incomplete (that's why unshelving isn't working in our case)

The function looks very overly complicated / hacky to determine the branch names.

**Comment by ITMATIC-HUSO** [comment]
I created a PR which changes the way the branches are determined.

Before the change, the two files were compared by character from end to start, which resulted in the issues. This happens for all projects ending with the same character, not just for projects containing a dot.

I changed it so it compares the folder structure of the two files against eachother, that way it won't strip out any folder names in case the characters match.
